### PR TITLE
scheduler: fix scatter range scheduler limit

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Closes https://github.com/tikv/pd/issues/3356

### What is changed and how it works?

This PR fixes the limit constriction for scatter range scheduler.

### Check List

Tests

- Unit test

Related changes

- Need to cherry-pick to the release branch

### Release note

- Fix the limit constriction of the scatter range scheduler

### CODE REVIEW

File: server/schedulers/scatter_range.go [hunk 1]
```diff
@@ -193,6 +193,15 @@ func (l *scatterRangeScheduler) EncodeConfig() ([]byte, error) {
 }
 
 func (l *scatterRangeScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
+	return l.allowBalanceLeader(cluster) || l.allowBalanceRegion(cluster)
+}
+
+func (l *scatterRangeScheduler) allowBalanceLeader(cluster opt.Cluster) bool {
+	// TODO: add metrics for leader limit
```
Comment (HunDunDM) [inline server/schedulers/scatter_range.go:200]: Will it continue to be added to this PR?

Comment (rleungx) [reply server/schedulers/scatter_range.go]: done

**Review by Yisaer — APPROVED** [review]
LGTM

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**